mqtt/mqtt-bridge/src/client.rs

```diff
@@ -194,20 +318,36 @@ impl<T: EventHandler> MqttClient<T> {
             .await
             .map_err(ClientConnectError::PollClientFailure)?
         {
-            //TODO: change the mqtt client to send an error back when the subscriotion fails instead of reconnecting and resending the sub
-            //right now it can't detect if the the broker doesn't allow to subscribe to the specific topic, but it will send a connect event
-            if let Event::NewConnection { reset_session: _ } = event {
-                return Ok(());
+            if let Event::SubscriptionUpdates(subscriptions) = event {
+                for subscription in subscriptions {
+                    match subscription {
+                        SubscriptionUpdateEvent::Subscribe(sub) => {
+                            subacks.remove(&sub.topic_filter);
+                        }
+                        SubscriptionUpdateEvent::SubscriptionRejectedByServer(sub) => {
+                            subacks.remove(&sub);
+                            error!("subscription rejected by server {}", sub);
+                        }
+
+                        SubscriptionUpdateEvent::Unsubscribe(_) => {}
+                    }
+                }
             }
         }
 
-        error!("failed to subscribe to topics");
-        Err(ClientConnectError::MissingSubacks(
-            subacks
-                .into_iter()
-                .map(std::string::ToString::to_string)
-                .collect(),
-        ))
+        if subacks.is_empty() {
+            debug!("successfully subscribed to topics");
+        } else {
+            error!(
+                "failed to receive expected subacks for topics: {0:?}",
+                subacks
+                    .iter()
+                    .map(std::string::ToString::to_string)
+                    .collect::<String>()
```

Comment:
nit: 
.collect()

Reply:
I get error: subacks.iter().map(ToString::to_string).collect()   cannot infer type for type parameter `B` declared on the associated function `collect`